lib/ChainwebDb/Types/Signer.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ImpredicativeTypes #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{-# OPTIONS_GHC -fno-warn-missing-signatures #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module ChainwebDb.Types.Signer where
+
+------------------------------------------------------------------------------
+import           Data.Aeson
+import           Data.Int
+import           Data.Text (Text)
+import           Database.Beam
+import           Database.Beam.AutoMigrate hiding (Table)
+import           Database.Beam.Backend.SQL hiding (tableName)
+import           Database.Beam.Backend.SQL.Row (FromBackendRow)
+import           Database.Beam.Backend.SQL.SQL92 (HasSqlValueSyntax)
+import           Database.Beam.Migrate
+import           Database.Beam.Postgres
+import           Database.Beam.Postgres.Syntax (PgValueSyntax)
+------------------------------------------------------------------------------
+import           ChainwebDb.Types.DbHash
+------------------------------------------------------------------------------
+
+newtype HexText = HexText { unHexText :: Text }
```

Comment:
Hmm, how does this differentiate from our keys which are generally hex? We actually can't assume what format hashes will be in for new schemes. I would suggest a `Signature` or similar to be more specific.

Reply:
Oh, that's a good point.  I hadn't thought about other signature schemes.  At the moment public keys keys are just plain `Text`, which I think makes sense because of the presence of the `scheme` field.  I was thinking about this type being generic and usable for any hex-encoded text but it probably does make sense to limit this to just signatures, which I think we can plan on always treating as hex.  I'll rename the data type.

Reply:
We can't assume anything, even hex.

Reply:
Got it.  I checked and we're actually not assuming hex anywhere here.